services/stickers/transactions.go

```diff
@@ -65,23 +71,99 @@ func (api *API) Buy(ctx context.Context, chainID uint64, txArgs transactions.Sen
 		return "", err
 	}
 
-	err = api.AddPending(chainID, packID)
+	api.txSignDetails = &txSigningDetails{
+		txType:        transactions.BuyStickerPack,
+		chainID:       chainID,
+		from:          txOpts.From,
+		txBeingSigned: tx,
+		packID:        packID,
+	}
+
+	signer := ethTypes.NewLondonSigner(new(big.Int).SetUint64(api.txSignDetails.chainID))
+	return signer.Hash(api.txSignDetails.txBeingSigned), nil
+}
+
+// Signs a transaction using appropriate local keystore file that is decrypted with provided password and returns the signature in hex format.
+// The tx that will be signed is the one that was prepared by the last call to PrepareTxForBuyingStickers method.
+func (api *API) SingPreparedTx(password string) (interface{}, error) {
+	if api.txSignDetails == nil {
+		return nil, errors.New("no tx to sign")
+	}
+
+	if api.txSignDetails.txType != transactions.BuyStickerPack {
+		return nil, errors.New("not supported tx type")
+	}
+
+	selectedAccount, err := api.accountsManager.VerifyAccountPassword(api.keyStoreDir, api.txSignDetails.from.Hex(), password)
+	if err != nil {
+		return nil, err
+	}
+
+	signer := ethTypes.NewLondonSigner(new(big.Int).SetUint64(api.txSignDetails.chainID))
+
+	hash := signer.Hash(api.txSignDetails.txBeingSigned)
+
+	signature, err := crypto.Sign(hash[:], selectedAccount.PrivateKey)
+
+	return types.EncodeHex(signature), err
+}
+
+// Sends a transaction using provided signature and returns the transaction hash.
+// Provided signature must be in hex format and must be the result of signing the tx that was prepared by the last call to PrepareTxForBuyingStickers method.
+func (api *API) SendPreparedTxWithSignature(ctx context.Context, signature string) (string, error) {
```

Comment:
Isn't it the same as in ens/api.go?

Reply:
It is not, different tx, different contract, different context.

Reply:
If you look at SendPreparedTxWithSignature in ens service, most of the code is identical, only the end differs.

Reply:
Aaa, sorry, different api.